spec/fixtures/test.pb.cr

```diff
@@ -15,7 +15,9 @@ struct Test
     required :sint64, :sint64, 13
     required :bool, :bool, 14
 
-    repeated :enum, SomeEnum, 15
+    # "enum" is a reserved keyword in
+    # crystal so we use "my_enum" for this one
+    repeated :my_enum, SomeEnum, 15
```

Comment:
Hm yes, I believe it may have to do with me calling the setters
from the constructor now (was necessary to get the new oneof-exclusive
behavior when doing e.g. `MyMessage.new(a: 1, b: 2, c: 3)`).

This produces code like `self.enum = @enum` which the compiler may not like.

Reply:
> perhaps adding the pretzel there would help

Agreed. If this code breaks with that change, that's a regression with no workaround. Definitely don't want that! 🙂

Reply:
Hmm, adding the `@` indeed fixes the enum issue.
But it also changes the behavior - the constructor then keeps the _first_ value instead of the last.

Before:
```
MyMsg.new(a: 1, b: 2, c: 3) # => { a: nil, b: nil, c: 3 }
```

After:
```
MyMsg.new(a: 1, b: 2, c: 3) # => { a: 1, b: nil, c: nil }
```

No idea what that is about, gonna take a closer look tomorrow.
(maybe it doesn't even matter anyway as this is not behavior that should be relied on 😬)

Reply:
(pushed an update with the enum fix and the above changed behavior for now)